rustls/src/crypto/aws_lc_rs/hpke.rs

```diff
@@ -896,66 +961,59 @@ mod tests {
-
-        // Opening the plaintext with the wrong opener should fail.
-        let mut sk_rm_prime = sk_rm.to_vec();
-        sk_rm_prime[0] = 0x00;
-        let mut opener_two = suite
-            .setup_opener(&enc, info, &HpkePrivateKey::from(sk_rm_prime))
-            .unwrap();
-        let open_res = opener_two.open(aad, &ct);
-        assert!(matches!(open_res.unwrap_err(), Error::Other(_)));
+        for suite in ALL_SUPPORTED_SUITES {
+            _ = format!("{suite:?}"); // HpkeAwsLcRs suites should be Debug.
+
+            // We should be able to generate a random keypair.
+            let (pk, sk) = suite.generate_key_pair().unwrap();
+
+            // Info value corresponds to the first RFC 9180 base mode test vector.
+            let info = &[
+                0x4f, 0x64, 0x65, 0x20, 0x6f, 0x6e, 0x20, 0x61, 0x20, 0x47, 0x72, 0x65, 0x63, 0x69,
+                0x61, 0x6e, 0x20, 0x55, 0x72, 0x6e,
+            ][..];
+
+            // We should be able to set up a sealer.
+            let (enc, mut sealer) = suite.setup_sealer(info, &pk).unwrap();
+
+            _ = format!("{sealer:?}"); // Sealer should be Debug.
+
+            // Setting up a sealer with an invalid public key should fail.
+            let bad_setup_res = suite.setup_sealer(info, &HpkePublicKey(vec![]));
+            assert!(matches!(bad_setup_res.unwrap_err(), Error::Other(_)));
+
+            // We should be able to seal some plaintext.
+            let aad = &[0xC0, 0xFF, 0xEE];
+            let pt = &[0xF0, 0x0D];
+            let ct = sealer.seal(aad, pt).unwrap();
+
+            // We should be able to set up an opener.
+            let mut opener = suite
+                .setup_opener(&enc, info, &sk)
+                .unwrap();
+            _ = format!("{opener:?}"); // Opener should be Debug.
+
+            // Setting up an opener with an invalid private key should fail.
+            let bad_key_res = suite.setup_opener(&enc, info, &HpkePrivateKey::from(vec![]));
+            assert!(matches!(bad_key_res.unwrap_err(), Error::Other(_)));
+
+            // Opening the plaintext should work with the correct opener and aad.
+            let pt_prime = opener.open(aad, &ct).unwrap();
+            assert_eq!(pt_prime, pt);
+
+            // Opening the plaintext with the correct opener and wrong aad should fail.
+            let open_res = opener.open(&[0x0], &ct);
+            assert!(matches!(open_res.unwrap_err(), Error::Other(_)));
+
+            // Opening the plaintext with the wrong opener should fail.
+            let mut sk_rm_prime = sk.secret_bytes().to_vec();
+            sk_rm_prime[10] = 0x00; // Corrupt a byte of the private key somewhere after potential zero padding.
+            let mut opener_two = suite
+                .setup_opener(&enc, info, &HpkePrivateKey::from(sk_rm_prime))
+                .unwrap();
+            let open_res = opener_two.open(aad, &ct);
+            assert!(matches!(open_res.unwrap_err(), Error::Other(_)));
```

Comment:
note to self; this test is flaky now that it's run for all suites with random keys instead of `DH_KEM_X25519_HKDF_SHA256_AES_128` w/ a fixed key - need to investigate why.

Reply:
I would suggest corrupting the key using `sk_rm_prime[10] ^= 0xff;` or similar -- just zeroing a byte in a uniformly random string will result in the same string 1 time out of 256.

Reply:
(assuming that is the issue!)

Reply:
>  just zeroing a byte in a uniformly random string will result in the same string 1 time out of 256.

:facepalm: Of course. That was silly of me. Fixed and no longer flaky in my local burst testing.